### PR TITLE
PMM-10059 integrate filters to table

### PR DESCRIPTION
**Pull request by filipmikes1**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.styles.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    `,
+    searchFields: css`
+      display: flex;
+      gap: 10px;
+      width: 100%;
+    `,
+    icon: css`
+      margin-top: 5px;
+      margin-bottom: 5px;
+    `,
```
Comment (fabio-silva) [inline public/app/percona/integrated-alerting/components/Table/Filter/Filter.styles.ts:46]: I see a lot of hard-coded paddings / margins / borders / ...
Can we try to use values from the theme as much as possible?

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+import { Messages } from './Filter.messages';
+
+export const Filter = ({ columns }: FilterProps) => {
+  const [openCollapse, setOpenCollapse] = useState(false);
+  const [openSearchFields, setOpenSearchFields] = useState(false);
+  const styles = useStyles2(getStyles);
+  const [queryParams, setQueryParams] = useQueryParams();
+
+  const searchColumnsOptions = useMemo(() => {
+    return buildSearchOptions(columns);
```
Comment (fabio-silva) [inline public/app/percona/integrated-alerting/components/Table/Filter/Filter.tsx:35]: We can immediately return here

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  }, [columns]);
+
+  const onFormChange = debounce((values: Record<string, any>) => {
+    const objForQueryParams = buildObjForQueryParams(columns, values);
+    setQueryParams(objForQueryParams);
+  }, DEBOUNCE_DELAY);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const initialValues = useMemo(() => getQueryParams(columns, queryParams), []);
+  console.log(initialValues);
```
Comment (fabio-silva) [inline public/app/percona/integrated-alerting/components/Table/Filter/Filter.tsx:45]:
```suggestion

```

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+      if (values[accessor]) {
+        if (values[accessor]?.value === ALL_VALUE || values[accessor] === ALL_VALUE) {
+          obj[accessor] = undefined;
+        } else {
+          obj[accessor] = values[accessor]?.value ? values[accessor]?.value.toString() : values[accessor].toString();
+        }
+      }
+    }
+  });
+  return obj;
```
Comment (fabio-silva) [inline public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts:53]: A bit like above, these ifs can be made simpler

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+};
+
+export const buildSearchOptions = (columns: ExtendedColumn[]) => {
+  const searchOptions = columns
+    .filter((value) => FilterFieldTypes.TEXT === value.type)
```
Comment (fabio-silva) [inline public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts:58]:
```suggestion
    .filter((value) => value.type === FilterFieldTypes.TEXT)
```

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+) => {
+  let result = false;
+  columns.forEach((column) => {
+    if (column.type === FilterFieldTypes.TEXT) {
+      if (queryParamsObj[SEARCH_INPUT_FIELD_NAME]) {
+        if (
+          column.accessor === queryParamsObj[SEARCH_SELECT_FIELD_NAME] ||
+          queryParamsObj[SEARCH_SELECT_FIELD_NAME] === ALL_VALUE
+        ) {
+          if (isTextIncluded(queryParamsObj[SEARCH_INPUT_FIELD_NAME], filterValue[column.accessor as string])) {
```
Comment (fabio-silva) [inline public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts:93]: Let's use consts again for these, it's harder to follow this way.
After having those, please see what can be made simpler here, 4 nested ifs seems a bit too much

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: public/app/percona/check/components/AllChecksTab/AllChecksTab.tsx [hunk 7]
```diff
@@ -137,11 +138,38 @@ export const AllChecksTab: FC = () => {
         Header: Messages.table.columns.status,
         accessor: 'disabled',
         Cell: ({ value }) => (!!value ? Messages.disabled : Messages.enabled),
+        type: FilterFieldTypes.RADIO_BUTTON,
+        label: 'Test',
```
[the remark this answers is not in the document]
Comment (filipmikes1) [reply public/app/percona/check/components/AllChecksTab/AllChecksTab.tsx]: removed

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.tsx (hunk 2 above)
Comment (filipmikes1) [reply public/app/percona/integrated-alerting/components/Table/Filter/Filter.tsx]: fixed

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.tsx [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+    }
+    if (numberOfParams > 2) {
+      setOpenCollapse(true);
+      setOpenSearchFields(true);
+    }
+    if (numberOfParams === 2) {
+      if (initialValues[SEARCH_INPUT_FIELD_NAME] && initialValues[SEARCH_SELECT_FIELD_NAME]) {
+        setOpenSearchFields(true);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (filipmikes1) [reply public/app/percona/integrated-alerting/components/Table/Filter/Filter.tsx:67]: But it is 3 different situations, I removed child ifs so it is more clear

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.tsx [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const showAdvanceFilter = useMemo(() => {
+    return isOtherThanTextType(columns);
```
[the remark this answers is not in the document]
Comment (filipmikes1) [reply public/app/percona/integrated-alerting/components/Table/Filter/Filter.tsx]: fixed

File: public/app/percona/check/components/AllChecksTab/AllChecksTab.tsx [hunk 10]
```diff
@@ -167,7 +195,15 @@ export const AllChecksTab: FC = () => {
       try {
         const checks = await CheckService.getAllChecks(generateToken(GET_ALL_CHECKS_CANCEL_TOKEN));
 
-        setChecks(checks);
+        setChecks(
+          checks.map((check) => {
+            if (check.disabled === undefined) {
+              return { ...check, disabled: false };
+            } else {
+              return check;
```
[the remark this answers is not in the document]
Comment (filipmikes1) [reply public/app/percona/check/components/AllChecksTab/AllChecksTab.tsx]: changed

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.tsx [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+  const [queryParams, setQueryParams] = useQueryParams();
+
+  const searchColumnsOptions = useMemo(() => {
+    return buildSearchOptions(columns);
+  }, [columns]);
+
+  const onFormChange = debounce((values: Record<string, any>) => {
+    const objForQueryParams = buildObjForQueryParams(columns, values);
+    setQueryParams(objForQueryParams);
+  }, DEBOUNCE_DELAY);
```
[the remark this answers is not in the document]
Comment (filipmikes1) [reply public/app/percona/integrated-alerting/components/Table/Filter/Filter.tsx]: Changed

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+  let obj: Record<string, any> = {
+    [SEARCH_INPUT_FIELD_NAME]: values[SEARCH_INPUT_FIELD_NAME],
+    [SEARCH_SELECT_FIELD_NAME]: values[SEARCH_SELECT_FIELD_NAME]?.value ?? values[SEARCH_SELECT_FIELD_NAME],
+  };
+  if (obj[SEARCH_INPUT_FIELD_NAME] && !obj[SEARCH_SELECT_FIELD_NAME]) {
+    obj[SEARCH_SELECT_FIELD_NAME] = ALL_VALUE;
+  }
+  if (!obj[SEARCH_INPUT_FIELD_NAME] && obj[SEARCH_SELECT_FIELD_NAME]) {
+    obj[SEARCH_SELECT_FIELD_NAME] = undefined;
+  }
```
[the remark this answers is not in the document]
Comment (filipmikes1) [reply public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts:32]: simplified

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts (hunk 4 above)
Comment (filipmikes1) [reply public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts:44]: simplified

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts (hunk 5 above)
Comment (filipmikes1) [reply public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts]: Fixed

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts (hunk 6 above)
Comment (filipmikes1) [reply public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts]: Used const, removed one if

File: public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+  filterValue: any,
+  queryParamsObj: { [key: keyof UrlQueryMap]: string },
+  filter: FilterFieldTypes
+) => {
+  let result: boolean[] = [];
+
+  columns.forEach((column) => {
+    const accessor = column.accessor as string;
+
+    if (column.type === filter) {
```
[the remark this answers is not in the document]
Comment (filipmikes1) [reply public/app/percona/integrated-alerting/components/Table/Filter/Filter.utils.ts]: it is type of field for example FilterFieldTypes.DROPDOWN. I changed name to filterFieldType so it is more clear.

**Review by fabio-silva — APPROVED** [review]

**Pull request merged**